integration_test/sql/subquery.exs

```diff
@@ -18,6 +18,7 @@ defmodule Ecto.Integration.SubQueryTest do
            TestRepo.all(from p in subquery(query), select: p)
   end
 
+  @tag :map_boolean_in_subquery
```

Comment:
@scouten MySQL has the same issue. They automatically convert `true` and `false` in the database driver into 1 and 0. If you don't want to do it on the adapter, you can do it on a pre-pass of the parameters on Ecto side, like Ecto's MySQL adapter does here: https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/adapters/mysql/connection.ex#L36-L45

We can also change the test to use another value rather boolean but I think the adapter would be benefit more if it could handle true and false.

We won't call the dumper because we consider true and false to be a native value.